Pspcl/Pspcl.Web/Controllers/ReportController.cs

```diff
@@ -67,5 +67,54 @@ public IActionResult StockOutReport()
 			}
 			return View();
 		}
+
+        [HttpGet]
+        public JsonResult FilteredStockInReport(DateTime? fromDate, DateTime? toDate)
+        {
+            try
+            {
+                var filteredstockInModels = _stockService.GetStockInModels();
+
+                if (fromDate.HasValue && toDate.HasValue)
+                {
+                    filteredstockInModels = filteredstockInModels.Where(s => s.Stock.GrnDate >= fromDate.Value && s.Stock.GrnDate <= toDate.Value)
```

Comment:
Filter should be OR, so that user can also filter records based on one date selection only.

Reply:
Both dates should be selected as per the requirement.